bin/xlifftranslate.js

```diff
@@ -114,55 +122,60 @@ function run() {
-            if (skipDifferent === true && source !== target) {
-              callback();
-              return;
-            }
-
-            text = replaceIgnoreTextWithPlaceholders(ignoreText, text, false);
-
-            translate.translate(text, lang).then(function (results) {
-              var translations = results[0];
-              var translation = Array.isArray(translations) ? translations[0] : translations;
-
-              translation = replaceIgnoreTextWithPlaceholders(ignoreText, translation, true);
-              if (verbose) {
-                console.log(`${locale}: ${text} => ${translation}`);
-              }
-              target.attr('xml:lang', locale);
-              target.attr('state', 'translated');
-              target.html(translation);
-              callback();
-            }).catch(function (err) {
-              if (err.code === 400) {
-                console.error(`Error: ${locale} is not a valid locale`);
-              } else {
-                console.error(err);
-              }
-              callback();
-            });
+        var source = node.find('source');
+        var target = node.find('target');
+        if (target.attr('state') === 'translated' ) {
+          callback();
+          return;
+        }
+
+        // skip if the caller wants to skip source/target that are unequal.
+        if (skipDifferent === true && source !== target) {
+          console.info("Skipped: " + text + " due to skipDifferent flag");
+          callback();
+          return;
+        }
+
+        text = replaceIgnoreTextWithPlaceholders(ignoreText, text, false);
+
+        translate.translate(text, lang).then(function (results) {
+          var translations = results[0];
+          var translation = Array.isArray(translations) ? translations[0] : translations;
+
+          translation = replaceIgnoreTextWithPlaceholders(ignoreText, translation, true);
+          if (verbose) {
+            console.log(`${locale}: ${text} => ${translation}`);
           }
-        } ());
-      });
-      runAsync.parallel(tasks, function () {
-        fs.writeFileSync(filePath, $.xml());
+          target.attr('xml:lang', locale);
+          target.attr('state', 'translated');
+          target.html(translation);
+          sleep.usleep(200)
+          callback();
+        }).catch(function (err) {
+          if (err.code === 400) {
+            console.error(`Error: ${locale} is not a valid locale`);
+          } else {
+            console.error(err);
+          }
+          callback();
+        });
+      }, 1);
+
+      // Translate - I.E. pass each node to the aync worker pool 'q' in order to run the translation function on the node.
+      $("trans-unit").each(function () {
+        var node = $(this);
+        q.push({'node': node}, function(err) {
+          // Precondition, node has been parsed, and xml file has been replaced with translations.
+          fs.writeFileSync(filePath, $.xml());
+        });
       });
```

Comment:
Code was minimally changed, but structure is different resulting in a confusing PR. 

Main difference are lines 169-175 above. This is the new way to add a task, pass data to the predefined processor function. I.E. - pass the node in and add the task to the queue.